lib/logstash/outputs/elasticsearch.rb

```diff
@@ -255,6 +255,11 @@ class LogStash::Outputs::ElasticSearch < LogStash::Outputs::Base
   # ILM policy to use, if undefined the default policy will be used.
   config :ilm_policy, :validate => :string, :default => DEFAULT_POLICY
 
+  # List extra HTTP's error codes that are considered valid to move the events into the dead letter queue.
+  # It's considered a configuration error to re-use the same predefined codes for success, DLQ or conflict.
+  # The option accepts a list of natural numbers corresponding to HTTP errors codes.
+  config :dlq_custom_codes, :validate => :number, :list => true, :default => []
```

Comment:
about the naming, I understand there are fixed codes 400 and 404, so name the changeable part as custom codes. As a user, I think `dlq_codes` is more approachable and easy to understand.

Reply:
I favor `dlq_custom_codes`, because I would assume `dlq_codes` to require passing a complete list to _override_ the list of codes we naturally route to DLQ, whereas the `custom` bit implies that it is in _addition_ to the natural routing.

Reply:
The intention was to inform the user that's not the full overriding list but an addition list to the default behavior.